deployment/lib/cf-url-rewriter.js

```diff
@@ -0,0 +1,59 @@
+const aws = require('aws-sdk');
+
+const s3 = new aws.S3({ apiVersion: '2006-03-01' });
+
+exports.handler = async (event, context) => {
+
+    const request = event.Records[0].cf.request;
+    request.uri = request.uri.replace(/^\/ci\/...\//, '\/')
+    callback(null, request);
+
+    // Below is the working in progress logic to get the max build number under a version.
+    // Hardcode the version to be 1.2.0 and bucket name to test-access-1-20 for demo purpose.
+    console.log('Received event:', JSON.stringify(event, null, 2));
+
+    // Get the object from the event and show its content type
+    const bucket = "test-access-1-20";
+
+    // Create the parameters for calling listObjects
+    var bucketParams = {
+        Bucket: bucket,
+        Prefix: '1.2.0/',
+        Delimiter: '/'
+    };
```

Comment:
Query with 'sort_by(Contents, &LastModified)[-1].Key' and a max 1, to get 1 latest item first, and just use that. https://stackoverflow.com/questions/31062365/get-last-modified-object-from-s3-using-aws-cli

Reply:
Thanks for sharing this! 

I have done some search in https://github.com/opensearch-project/opensearch-build/issues/1492

Based on my research, AWS CLI is different from AWS SDK. Also there is no such query thing in S3 SDK https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html This is the reason why I choose to use `prefix` and `delimiter`. Happy to learn if there is more elegant way to solve this problem.

Reply:
I see. The problem is that the current code as is doesn't seem to work past a page of data. Paginating on every request is probably something we can live with, but it's not ideal.

I wonder whether we're better off writing a `latest.build` file into S3 and loading that here?

Reply:
I still want pagination then, because we will not remember that this is a limitation of our redirector when we hit 1000 in the worst possible moment, and _everything_ will break.

Reply:
Let me experiment a bit to see if we can extend the file name `latest.build` to `build.{version}.{latest build number}` so that the result will always be 1 (which will get rid of pagination). This may require potentially updating Jenkins to upload such object after regular artifact uploading or introducing another Lambda to add such once distribution bucket has new update.

Reply:
If you're going to be uploading a file, just upload a metadata file to the bucket/folder with a bunch of information ala maven, e.g. `$BUCKET/distribution-build-opensearch/1.2.4/index.yml` that contains `latest: xyz`.

Reply:
One (minor) concern I would love to raise here is that, we may need to consider the completeness of the uploaded artifacts. e.g. one major version may include a lot of files, once the first file is uploaded to S3, the version number will be present when you list the objects in the bucket, however during the upload time, some of the artifact files may not been uploaded to S3 yet. Or even the upload process may crash due to some network issue. Then you may serve incomplete contents to users during that period of time.

Maybe it is better for the build system to update the version mapping in the `latest.build` file, or the build system can create a flag so this URL rewriter can know the latest version upload is complete.  Then we can make sure we always serve complete build artifacts.

Reply:
This is a great catch of the PR as implemented @zengyan-amazon. By iterating through folders you may be getting a build in-progress of being uploaded.

Reply:
Thanks all for the input! 

Checked a bit about the existing artifacts uploading logic https://github.com/opensearch-project/opensearch-build/blob/main/vars/uploadArtifacts.groovy#L12 Looks like that we are already allowing the phased uploading of artifacts. (The artifacts are not ready simultaneously for the build number based url ) Furthermore, it seems we are not handling the uploading errors. Just wondering whether the existing approach is chosen because we are allowing some flexibility for non-release endpoints.

Reply:
The logic to add `index.json` is completed in this https://github.com/opensearch-project/opensearch-build/pull/1622